pkg/dockercompose/dockercompose.go

```diff
@@ -121,6 +121,33 @@ func Load(composePath string, overridePaths, services []string) (types.Project,
 			filepath.Base(filepath.Dir(composePath)), svc.Name)
 	}
 
+	// Convert build contexts to absolute paths, set the default value for the
+	// dockerfile field, and validate that the dockerfile exists.
+	for svcIdx, svc := range cfgPtr.Services {
+		if svc.Build == nil {
+			continue
+		}
+
+		cfgPtr.Services[svcIdx].Build.Context = filepath.Join(filepath.Dir(composePath), svc.Build.Context)
+		if svc.Build.Dockerfile == "" {
+			cfgPtr.Services[svcIdx].Build.Dockerfile = "Dockerfile"
```

Comment:
Buildkit doesn't actually need this. If Dockerfile is `""`, it will default to `"Dockerfile"` correctly. But this doesn't hurt.

Reply:
Ah plus we need it for below, so we might as well.